torch/optim/nadam.py

```diff
@@ -262,8 +261,8 @@ def _single_tensor_nadam(params: List[Tensor],
             param.addcdiv_(exp_avg, denom, value=1.0)
         else:
             denom.add_(eps)
-            param.addcdiv_(grad, denom, value=-lr * (1. - mu) / (1. - mu_product.item()))
-            param.addcdiv_(exp_avg, denom, value=-lr * mu_next / (1. - mu_product_next.item()))
+            param.addcdiv_(grad, denom, value=-lr * (1. - mu) / (1. - mu_product))
```

Comment:
TODO: we can remove most (if not all) of the if statement here now that this side doesn't use `.item()` either.

Reply:
Dynamo doesn't like the `value` argument since the scalars aren't  traceable, is it okay if I go with the then clause of the `if`? Similar question with the addcdiv_ in Adam. I can also post numbers showing that Adam single tensor with dynamo + inductor is faster than the multitensor version if this helps with justification

Reply:
Resolved offline, I have this PR here: https://github.com/pytorch/pytorch/pull/90227 should fix this.